lib/rubocop/cop/rails/left_join.rb

```diff
@@ -0,0 +1,69 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Rails
+      # This cop looks for both .joins("LEFT JOIN ...") and
+      # .joins("LEFT OUTER JOIN ...").
+      # It proposes to use either the .left_joins("...") method
+      # or the .left_outer_joins("...") one accordingly.
+      # These methods were introduced in Rails 5.0.
+      #
+      # @example
+      #  # bad
+      #  User.joins('LEFT JOIN emails ON user.id = emails.user_id')
+      #
+      #  # good
+      #  User.left_joins(:emails)
+      #
+      #
+      #  # bad
+      #  User.joins('LEFT OUTER JOIN emails ON user.id = emails.user_id')
+      #
+      #  # good
+      #  User.left_outer_joins(:emails)
+      #
+      class LeftJoin < Cop
+        extend TargetRailsVersion
+
+        minimum_target_rails_version 5.0
+
+        MSG = 'Use `%<prefer>s` instead of `%<current>s`.'.freeze
+
+        PREFERRABLE_QUERIES = {
+          'left join'       => '.left_joins(:model)',
+          'left outer join' => '.left_outer_joins(:model)'
```

Comment:
Could the `:model` here be inferred by the query to have a better error message?